modules/api/pkg/handler/v2/provider/gcp.go

```diff
@@ -106,6 +164,278 @@ func DecodeGCPSubnetworksNoCredentialReq(c context.Context, r *http.Request) (in
 	return req, nil
 }
 
+// Validate checks that ServiceAccount and Credentials aren't empty
+func (req GCPCommonReq) Validate() error {
+	if len(req.ServiceAccount) == 0 && len(req.Credential) == 0 {
+		return fmt.Errorf("GCP credentials cannot be empty")
+	}
+	return nil
+}
+
+func getSAFromPreset(ctx context.Context,
+	userInfoGetter provider.UserInfoGetter,
+	presetProvider provider.PresetProvider,
+	presetName string,
+	projectID string,
+) (string, error) {
+	userInfo, err := userInfoGetter(ctx, projectID)
+	if err != nil {
+		return "", common.KubernetesErrorToHTTPError(err)
+	}
+	preset, err := presetProvider.GetPreset(ctx, userInfo, &projectID, presetName)
+	if err != nil {
+		return "", utilerrors.New(http.StatusInternalServerError, fmt.Sprintf("can not get preset %s for user %s", presetName, userInfo.Email))
+	}
+	credentials := preset.Spec.GKE
+	if credentials == nil {
+		return "", fmt.Errorf("gke credentials not present in the preset %s", presetName)
+	}
+	return credentials.ServiceAccount, nil
+}
+
+func ListProjectGCPDiskTypes(presetProvider provider.PresetProvider, userInfoGetter provider.UserInfoGetter, withProject bool) endpoint.Endpoint {
```

Comment:
I know we discussed this with the `withProject` variable so this is my bad, but in this specific case you can simplify your code by dropping the `withProject` boolean and just always assume a project-aware request is passed, simply because the old API endpoints for GCP are in `/v1/` and you needed to copy over the endpoint functions anyway.

Sorry for the confusion, I know I showed you something _with_ this variable.